### ISSUE TITLE
gmock has moved, compilation of submodule protobuf fails

### ISSUE THREAD

**Issue by philippbayer** [wtsi-svi/ReadServer#14]
When Google Mock is not present during installation the submodule protobuf tries to pull Google Mock from the web and 404s since Google Mock moved to GitHub.

The solution is to update the protobuf submodule version (the index of the superproject points to v3.0.0-beta-2, the newest tag is v3.1.0)

Alternatively, put this line into the install_dependencies.sh file:

    git submodule foreach git pull origin master

This will force an update of all submodules during installation, could break things in the future.

I learned something about git submodules today! I just assumed that `git submodule update` gets the newest commit, but that's not the case, see [here](http://stackoverflow.com/questions/5828324/update-git-submodule-to-latest-commit-on-origin)

**Comment by Zhicheng-Liu** [comment]
@philippbayer Thank you very much for bringing this issue to our attention. I have updated the Protobuf to v3.1.0.

I prefer to update this submodule only as, if I remember correctly, one of the other submodules has recent API changes that will break our current implementation.

**Comment by philippbayer** [comment]
Thank you very much for your quick reaction! However, this introduced another minor problem. 'install_dependencies.sh' worked fine, but the main Makefile didn't work out:

```
../../include/service/readserver.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
 #error regenerate this file with a newer version of protoc.
  ^
```

Adding the line ` syntax = "proto2"; ` to the top of `./src/service/readserver.proto` and rerunning protoc like this:

    ./submodules/protobuf/src/protoc ./src/service/readserver.proto --cpp_out=src/service/

Seems to have fixed it? At least `make` doesn't complain anymore.

By the way, maybe something for the README: with linuxbrew I had to do 

    brew install gcc cmake libtool zlib texinfo texi2html automake

to the all dependencies in my ~/.linuxbrew

**Comment by Zhicheng-Liu** [comment]
@philippbayer Thank you very much for reporting the issue. I believe this will be fixed by bbfc191 together with changes made to address issue #15.

In terms of some make tools and other libraries you mentioned which may be missing from some OSs, they are listed in the README already:
https://github.com/wtsi-svi/ReadServer/blame/master/README.md#L50-L68

Once again, thank you very much for your help!
